ansible/roles/vm_set/tasks/add_ceos_list.yml

```diff
@@ -6,6 +6,28 @@
     fp_mtu:       "{{ fp_mtu_size }}"
     max_fp_num:   "{{ max_fp_num }}"
 
+- name: Check if cEOS image exists or not
+  docker_image_info:
+    name:
+      - "{{ ceos_image_orig }}"
+  become: yes
+  register: ceos_stat
+
+- include_vars: vars/azure_storage.yml
+
+- name: Download cEOS image
+  become: yes
+  get_url:
+    url: "{{ vm_images_url }}/{{ ceos_image_filename }}?{{ ceosimage_saskey }}"
+    dest: "{{ root_path }}/images/{{ ceos_image_filename }}"
+    timeout: 1800
+  when: ceos_stat.images | length == 0
+
+- name: Import cEOS image
+  become: yes
+  shell: "docker import {{ root_path }}/images/{{ ceos_image_filename }} {{ ceos_image_orig }}"
+  when: ceos_stat.images | length == 0
```

Comment:
not all developer has the docker registry setup. I think we should still maintain the existing option.

Reply:
I don't understand this request. Could you elaborate?

Docker registry is mandatory for the SONiC test setup. Currently we pull PTF image from docker registry when run `add-topo`.  So, for a running test setup, there is always a docker registry there.

For cEOS docker image, no docker registry is involved. If cEOS docker image is not in local test server, then download it from HTTP server and import it locally. If it is already in local server (previously imported), then skip downloading and importing.

Reply:
maybe we can check if ceosimage_saskey is defined or not. if not defined, then do not download since user may have other ways to download that.

Reply:
Updated. The `ceosimage_saskey` is set to `use_own_value`. It is always defined and is not a good option for skipping image downloading. I added another flag `skip_ceos_image_downloading`  in `sonic-mgmt/ansible/group_vars/all/ceos.yml` for skipping image downloading. It follows the same logic of downloading EOS VM image: https://github.com/Azure/sonic-mgmt/blob/master/ansible/group_vars/vm_host/main.yml#L5

Document section https://github.com/Azure/sonic-mgmt/blob/master/ansible/doc/README.testbed.VsSetup.md#use-ceos-image-experimental is also updated.